sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/FetchSubscriberDbConnector.groovy

```diff
@@ -0,0 +1,99 @@
+package life.qbic.samplenotificator.datasource.notification.create
+
+import groovy.util.logging.Log4j2
+import life.qbic.business.exception.DatabaseQueryException
+import life.qbic.business.subscription.fetch.FetchSubscriberDataSource
+import life.qbic.business.subscription.Subscriber
+import life.qbic.datamodel.samples.Status
+import life.qbic.samplenotificator.datasource.database.ConnectionProvider
+import life.qbic.samplenotificator.datasource.database.DatabaseSession
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+
+/**
+ * <h1>A database connector to retrieve information for sending notifications</h1>
+ *
+ * <p>This connector reads the notification queue table and finds the associated people that subscribed for a notification</p>
+ *
+ * @since 1.0.0
+ *
+*/
+@Log4j2
+class FetchSubscriberDbConnector implements FetchSubscriberDataSource{
+    private ConnectionProvider connectionProvider
+
+    FetchSubscriberDbConnector(ConnectionProvider connectionProvider){
+        this.connectionProvider = connectionProvider
+    }
+
+    @Override
+    Map<String, Status> getUpdatedSamplesForDay(LocalDate day){
+        Instant startOfDay = day.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant()
+        println startOfDay.toString()
+        Instant endOfDay = day.plusDays(1).atStartOfDay().minusNanos(1).atZone(ZoneId.systemDefault()).toInstant()
+        println endOfDay.toString()
+        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE arrival_time BETWEEN ? AND ?"
+        Map<String, Status> foundNotifications = new HashMap<>()
+
+        try{
+            Connection connection = connectionProvider.connect()
+
+            connection.withCloseable {
+                PreparedStatement preparedStatement = it.prepareStatement(sqlQuery)
+                preparedStatement.setTimestamp(1, Timestamp.from(startOfDay))
+                preparedStatement.setTimestamp(2, Timestamp.from(endOfDay))
+                preparedStatement.execute()
+
+                def resultSet = preparedStatement.getResultSet()
+                while (resultSet.next()) {
+                    String sampleCode = resultSet.getString("sample_code")
+                    Status status = Status.valueOf(resultSet.getString("sample_status"))
+                    foundNotifications.put(sampleCode,status)
+                }
```

Comment:
I found that when a samples has multiple status changes on the same day, only the last on will be stored (Since we only store one sample to one status and this is unique). I will clarify what behaviour we want to have here

Reply:
@Steffengreiner do you think the last (most resent) status would make sense here?

Reply:
and then we would neglect the previous status

Reply:
![image](https://user-images.githubusercontent.com/11536497/130606495-9e339e56-6411-4e8b-a4d2-1ab6e97cf84f.png)
maybe this is important here

Reply:
I think it makes sense for tracking the history, but for the user the last status is most important I would think

Reply:
I'd also agree that the latest status makes the most sense for an email notification tool, since AFAIK the user is concerned on the current status of his samples. 
I'd argue that if this is relevant later this can be added in a seperate PR so this is not blocked again.